### PR TITLE
Adapt API changes in Agenda v2

### PR DESCRIPTION
**Pull request by 2bigO**
WIP on #20

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ const agenda = new Agenda({
   }
 });
 
-const jobsReady = promisify(agenda.on).bind(agenda)('ready')
+const jobsReady = Promise.resolve(agenda._ready)
```
Comment (OmgImAlexis) [inline src/index.js]: Doesn't `_ready` already return a promise?

Comment (2bigO) [reply src/index.js]: Right! 🙈

File: package.json [hunk 2]
```diff
@@ -40,31 +40,31 @@
   },
   "homepage": "https://github.com/agenda/agenda-rest#README",
   "devDependencies": {
-    "@babel/cli": "^7.0.0-beta.40",
-    "@babel/core": "^7.0.0-beta.40",
-    "@babel/preset-env": "^7.0.0-beta.40",
-    "@babel/register": "^7.0.0-beta.40",
+    "@babel/cli": "^7.0.0-beta.55",
```
Comment (koresar) [inline package.json]: They've just released v7 stable. `npm view @babel/cli`
Might be a good idea to upgrade to the latest.

Comment (2bigO) [reply package.json]: Great, will do right now 👍

**Review by simison — COMMENTED** [review]
Gave this a spin!

Code looks good, although I'm not super familiar with this codebase.

`npm run lint` complains about a few things:
```
  test.js:75:1
  ⚠  75:1  Unexpected todo comment.                no-warning-comments

  src/index.js:1:1
  ✖   1:1  Parsing error: Unexpected token import

  src/job.js:1:1
  ✖   1:1  Parsing error: Unexpected token import

  src/util.js:1:1
  ✖   1:1  Parsing error: Unexpected token import

  1 warning
  3 errors
```

`npm test` has one failure:
```
(node:17268) DeprecationWarning: current URL string parser is deprecated, and will be removed in a future version. To use the new parser, pass option { useNewUrlParser: true } to MongoClient.connect.
agenda-rest app running
test app running
(node:17268) DeprecationWarning: collection.count is deprecated, and will be removed in a future version. Use collection.countDocuments or collection.estimatedDocumentCount instead
(node:17268) DeprecationWarning: collection.update is deprecated. Use updateOne, updateMany, or bulkWrite instead.
  <-- POST /api/job
{ Error: expected request body to match {name, url}
    at Object.job [as check] (/Users/mikael/Documents/Agenda/agenda-rest/dist/webpack:/[name]/src/job.js:9:11)
    at jobOperation (/Users/mikael/Documents/Agenda/agenda-rest/dist/webpack:/[name]/src/job.js:155:15)
    at /Users/mikael/Documents/Agenda/agenda-rest/dist/webpack:/[name]/src/index.js:36:20
    at process._tickCallback (internal/process/next_tick.js:68:7) expose: true, statusCode: 400, status: 400 }
  --> POST /api/job 400 48ms 49b
  <-- POST /api/job
{ Error: A job named "foo" already exist
    at jobs.count.then.count (/Users/mikael/Documents/Agenda/agenda-rest/dist/webpack:/[name]/src/job.js:19:13)
    at process._tickCallback (internal/process/next_tick.js:68:7) expose: true, statusCode: 400, status: 400 }
  --> POST /api/job 400 35ms 38b
  <-- PUT /api/job/fooWrong
{ Error: Did not find a job named "fooWrong"
    at jobs.count.then.count (/Users/mikael/Documents/Agenda/agenda-rest/dist/webpack:/[name]/src/job.js:19:13)
    at process._tickCallback (internal/process/next_tick.js:68:7) expose: true, statusCode: 400, status: 400 }
  --> PUT /api/job/fooWrong 400 2ms 42b
  <-- PUT /api/job/foo
  --> PUT /api/job/foo 200 8ms 11b
  <-- POST /api/job/now
  --> POST /api/job/now 200 19ms 21b
  <-- POST /foo
foo now invoked! 1 of 1
  --> POST /foo 200 3ms 15b
  <-- POST /api/job/every
  --> POST /api/job/every 200 20ms 28b
  <-- POST /foo
foo every invoked! 1 of 3
  --> POST /foo 200 2ms 17b
  <-- POST /foo
foo every invoked! 2 of 3
  --> POST /foo 200 0ms 17b
  <-- POST /foo
foo every invoked! 3 of 3
  --> POST /foo 200 1ms 17b
  <-- POST /api/job/once
  --> POST /api/job/once 200 2ms 22b
  <-- POST /foo
foo once invoked! 1 of 1
  --> POST /foo 200 1ms 16b
  <-- DELETE /api/job/foo
(node:17268) DeprecationWarning: collection.remove is deprecated. Use deleteOne, deleteMany, or bulkWrite instead.
  --> DELETE /api/job/foo 200 9ms 46b

  7 passed
  1 failed

  POST /api/job succeeds when a job is specified

  /Users/mikael/Documents/Agenda/agenda-rest/dist/webpack:/[name]/test.js:38

  Difference:

  - 400
  + 200
```

Build, dev and start commands run fine.

When looking at `http://localhost:4040/api/job`, I see:
```json
[
  {
    "_id": "5a765270b6a4a07126e4faf3",
    "name": "foo",
    "url": "http://localhost:4042/foo",
    "method": null,
    "callback": null
  }
]
```
...not sure if that URL with funny port is something of a concern. I did run tests before this so maybe it's a leftover from that or from some previous runs I've done?

When shutting down the server with `CMD`+`C`, it seems to get stuck to _"Shutting down gracefully..."_ and never reaches `process.exit(0)` — hitting cmd+c another time kills with with `Terminated: 15`

**Comment by 2bigO** [comment]
Thanks @simison, the tests are a sequence of creating and then deleting the `foo` job. My guess is if you run `npm test` a second time, the failing test would pass.

You pointed exactly at two of my headaches, I'm not sure why and when ESLint's behaviour changed and now complains about `import`! And as you can see, it doesn't error on Travis, which is weird. Also I relocated the code for `CTRL+C` and moved it to `cli.js`, that solved things on my Linux Dev machine, though I have the exact issue on my Windows Dev env. Inputs on these two issues are very much appreciated. Alternatively, we could merge and track these on separate issues and PRs. Thank you very much for your time and consideration.

**Comment by koresar** [comment]
To fix ESLint I would assume you need to update the `"xo":` property in package.json. Probably add this ESLint config:
```
  "xo": {
    "parserOptions": {
      "ecmaVersion": 2017
    },
```

**Comment by OmgImAlexis** [comment]
You likely need to enable [`esnext`](https://github.com/xojs/xo#esnext).

**Comment by 2bigO** [comment]
@koresar @OmgImAlexis thanks for the suggestions. These values are supposedly the defaults for XO. I updated to latest and the issue resolved on my dev machine, also Travis is still not complaining. Maybe it was an issue with the XO release back then? Thanks for pointing out those configs 👍

**Comment by OmgImAlexis** [comment]
@keyvan-m-sadeghi I use xo for every project I work on. The default are good but without things like `esnext` you don't have all ES6/7 features so it's always good to enable.

**Comment by 2bigO** [comment]
@OmgImAlexis I meant it seems that the [esnext](https://github.com/xojs/xo#esnext) option is enabled by default (docs says: Default: `true`), you suggest we still need to set it?

**Comment by OmgImAlexis** [comment]
@keyvan-m-sadeghi ah, that's changed since I last looked. No need. 👍

**Pull request merged**